runtime/src/bank/partitioned_epoch_rewards/sysvar.rs

```diff
@@ -25,17 +25,28 @@ impl Bank {
         total_rewards: u64,
         distributed_rewards: u64,
         distribution_starting_block_height: u64,
+        num_partitions: u64,
+        total_points: u128,
     ) {
         assert!(self.is_partitioned_rewards_code_enabled());
 
         assert!(total_rewards >= distributed_rewards);
 
+        // If Bank::parent() does not exist, as in slot 0, use Hash::default()
+        // for the parent_blockhash.
+        let parent_blockhash = self
+            .parent()
+            .map(|bank| bank.last_blockhash())
+            .unwrap_or_default();
```

Comment:
You should be able to use `self.last_blockhash()` here because this is a new bank and it's last blockhash is always the parent's last blockhash.

Reply:
Oooh, nice one. Thanks, that's much better.

Reply:
Oh actually I meant you could just do:
```
let parent_blockhash = self.last_blockhash();
```

Reply:
Oh of course, because we haven't started processing yet. Duh. Let me make sure the tests confirm, but that would be much cleaner.

Reply:
Yep, looks good. Updated that last commit.